packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -212,3 +237,46 @@ export async function destroyStack(options: DestroyStackOptions) {
   }
   return;
 }
+
+async function getDeployedTemplate(cfn: aws.CloudFormation, stackName: string): Promise<{ template: any, stackId: string } | undefined> {
+  const stackId = await getStackId(cfn, stackName);
+  if (!stackId) {
+    return undefined;
+  }
+
+  const template = await readCurrentTemplate(cfn, stackName);
+  return { stackId, template };
+}
+
+export async function readCurrentTemplate(cfn: aws.CloudFormation, stackName: string) {
+  try {
+    const response = await cfn.getTemplate({ StackName: stackName, TemplateStage: 'Original' }).promise();
+    return (response.TemplateBody && deserializeStructure(response.TemplateBody)) || {};
+  } catch (e) {
+    if (e.code === 'ValidationError' && e.message === `Stack with id ${stackName} does not exist`) {
+      return {};
+    } else {
+      throw e;
+    }
+  }
+}
+
+async function getStackId(cfn: aws.CloudFormation, stackName: string): Promise<string | undefined> {
+  try {
+    const stacks = await cfn.describeStacks({ StackName: stackName }).promise();
+    if (!stacks.Stacks) {
+      return undefined;
+    }
+    if (stacks.Stacks.length !== 1) {
+      return undefined;
+    }
```

Comment:
out of curiosity, when does this happen?

Reply:
Never, just being defensive (and pleasing the typescript compiler).